game-core/src/main/java/games/strategy/net/AbstractServerMessenger.java

```diff
@@ -304,13 +289,11 @@ public void notifyPlayerLogin(final String uniquePlayerName, final String mac) {
   }
 
   private void notifyPlayerRemoval(final INode node) {
-    synchronized (cachedListLock) {
-      playersThatLeftMacsLast10.put(node.getName(), cachedMacAddresses.get(node.getName()));
-      if (playersThatLeftMacsLast10.size() > 10) {
-        playersThatLeftMacsLast10.remove(playersThatLeftMacsLast10.entrySet().iterator().next().toString());
-      }
-      cachedMacAddresses.remove(node.getName());
+    playersThatLeftMacsLast10.put(node.getName(), cachedMacAddresses.get(node.getName()));
+    if (playersThatLeftMacsLast10.size() > 10) {
+      playersThatLeftMacsLast10.remove(playersThatLeftMacsLast10.entrySet().iterator().next().toString());
```

Comment:
I feel like this is a NO-OP:
I believe `entrySet()` should probably get replaced with `keySet()`, but in this case `playersThatLeftMacsLast10.entrySet().iterator().remove()` might be used as well.